fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/VersionKeyExpression.java

```diff
@@ -107,7 +108,7 @@ public RecordMetaDataProto.KeyExpression toKeyExpression() {
 
     @Nonnull
     @Override
-    public Value toValue(@Nonnull final CorrelationIdentifier baseAlias, @Nonnull final List<String> fieldNamePrefix) {
+    public Value toValue(@Nonnull final CorrelationIdentifier baseAlias, @Nonnull final Type baseType, @Nonnull final List<String> fieldNamePrefix) {
```

Comment:
if in the caller you always have the `baseQuantifier` you might as well hand in that one as it holds both the type and the alias. I think I started with just the alias but then later on changed quite a few places to take a quantifier but apparently not all.

Reply:
I took a look at this, and there is one case where we actually don't have the base quantifier, namely `ScalarTranslationVisitor`'s visit method for `KeyExpressionWithValue`:

https://github.com/FoundationDB/fdb-record-layer/blob/c90ef5e336e384d94b35e351d72cf7b8033bad8c/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/ScalarTranslationVisitor.java#L138-L141

The `ScalarTranslationVisitor` only has type and alias information. I did briefly take a look at if I could modify it to use a quantifier, but it ended up being a little bit more tricky than I'd hoped. It usually uses the `Quantifier.current()` alias as its basis rather than using a real quantifier.

I added a default variant to this `toValue` method that takes a quantifier while delegating to the other method, and I updated the call sites that I could to use it.

Reply:
okay bummer